geoips/plugins/modules/procflows/config_based.py

```diff
@@ -991,7 +991,7 @@ def call(fnames, command_line_args=None):
         )
         bg_xobjs = bg_reader_plugin(bg_files, metadata_only=True, **bg_reader_kwargs)
         prod_plugin = products.get_plugin(
-            bg_xobjs["METADATA"].source_name,
+            get_source_name(bg_xobjs),
```

Comment:
There should still be a top level "source_name" in the metadata, even if individual source_file_attributes are defined.  Top level source name could be "multiple" or something if not all included files have the same source.

I expect the readers should set the top level source_name as well as the individual source names for each file so this should not have to change.

Reply:
I refactored to keep source_name (and other required attributes) top level as you suggested.

Reply:
This change was reverted to keep source name at the top level.

Reply:
Thanks!